server/storage/storage_impl.go

```diff
@@ -188,118 +306,325 @@ func (s *metaStorageImpl) PutSchemas(ctx context.Context, clusterID uint32, sche
 		key := makeSchemaKey(clusterID, schema.Id)
 		value, err := proto.Marshal(schema)
 		if err != nil {
-			return ErrParsePutSchemas.WithCausef("proto parse failed, clusterID:%d, schemaID:%d, err:%s", clusterID, schema.Id, err)
+			return ErrParsePutSchemas.WithCausef("fail to parse proto, clusterID:%d, schemaID:%d, err:%v", clusterID, schema.Id, err)
```

Comment:
```suggestion
			return ErrEncodeSchema.WithCausef("fail to encode schema, clusterID:%d, schemaID:%d, err:%v", clusterID, schema.Id, err)
```

Actually, there is no need to define a special error for every operation method, e.g. `ErrParsePutSchemas`, `ErrParseGetSchemas`, and only one error called `ErrParseSchema` is enough.

BTW, here is actually a decoding error rather than parsing/encoding.

This style distributes across the whole file and you may need to fix them all.